gemma-core/src/main/java/ubic/gemma/core/ontology/OntologyService.java

```diff
@@ -185,10 +195,10 @@ Collection<CharacteristicValueObject> findExperimentsCharacteristicTags( String
     void reindexAllOntologies();
 
     /**
-     * Reinitialize all the ontologies "from scratch". This is necessary if indices are old etc. This should be
+     * Reinitialize (and reindex) all the ontologies "from scratch". This is necessary if indices are old etc. This should be
      * admin-only.
      */
-    void reinitializeAllOntologies();
+    void reinitializeAndReindexAllOntologies();
```

Comment:
We shouldn't force re-index in general. baseCode has some capability to detect if the ontology actually changed and require a reindexing. Maybe adding a forceReindex flag would be more appropriate?